### PR TITLE
CY-2363 Add a "validate blueprint" REST endpoint

### PR DESCRIPTION
**Pull request by mateumann**
Add `BlueprintsClient.validate()` method in `cloudify_rest_client` module for handling blueprints validation.  As validation is quite similar to upload, there is some refactoring as well.

### CODE REVIEW

File: cloudify_rest_client/blueprints.py [hunk 1]
```diff
@@ -225,6 +262,48 @@ def upload(self,
...
+        """
+        tempdir = tempfile.mkdtemp()
+        try:
+            tar_path = utils.tar_blueprint(path, tempdir)
+            if not skip_size_limit and os.path.getsize(tar_path) > 30000000:
+                raise Exception('Blueprint folder exceeds 30 MB, '
+                                'move some resources from the blueprint '
+                                'folder to an external location or upload'
+                                ' the blueprint folder as a zip file.')
+            application_file = os.path.basename(path)
```
Comment (mabuaisha) [inline cloudify_rest_client/blueprints.py]: The above logic is similar to what we have in upload method, cannot we this logic to share method called `_validte_blueprint_size` for example ?

File: cloudify_rest_client/blueprints.py [hunk 2]
```diff
@@ -225,6 +262,48 @@ def upload(self,
...
+        Validates a blueprint with Cloudify's manager.
+
+        :param path: Main blueprint yaml file path.
+        :param entity_id: Id of the uploaded blueprint.
+        :param visibility: The visibility of the blueprint, can be 'private',
+                           'tenant' or 'global'.
+        :param progress_callback: Progress bar callback method
+        :param skip_size_limit: Indicator whether to check size limit on
+                           blueprint folder
+        :return: Created response.
```
Comment (tehasdf) [inline cloudify_rest_client/blueprints.py]: that's a lie ;)

File: cloudify_rest_client/blueprints.py [hunk 3]
```diff
@@ -225,6 +262,48 @@ def upload(self,
...
+
+        Blueprint path should point to the main yaml file of the response
+        to be uploaded. Its containing folder will be packed to an archive
+        and get uploaded to the manager.
+        Validation is basically an upload without storage part being done.
```
Comment (tehasdf) [inline cloudify_rest_client/blueprints.py]: s/without storage/without **the** storage/

File: cloudify_rest_client/blueprints.py (hunk 1 above)
Comment (mateumann) [reply cloudify_rest_client/blueprints.py]: You are right; fixed that.

File: cloudify_rest_client/blueprints.py (hunk 2 above)
Comment (mateumann) [reply cloudify_rest_client/blueprints.py]: that's true ;)

File: cloudify_rest_client/blueprints.py (hunk 3 above)
Comment (mateumann) [reply cloudify_rest_client/blueprints.py]: ok

**Pull request merged**